### PR TITLE
fix: deps for cvxpy

### PR DESCRIPTION
**Pull request by virajvchaudhari**
*Issue #, if available:*

*Description of changes:*
Added ecos dependency for running tests locally without any issues for Randomness Generation notebook. This dependency was recently updated in NBIs. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ dependencies:
       - amazon-braket-sdk
       - dask==2.30.0
       - dwave-ocean-sdk==3.3.0
+      - ecos==2.0.7.post1
```
Comment (kshitijc) [inline environment.yml:39]: We should also pin cvxpy. Also can we remove the pip install from the randomness generation notebook then, since it won't be needed ?

Comment (virajvchaudhari) [reply environment.yml:39]: Yes, let me check if that causes any issues with other dependencies.

File: examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb [hunk 2]
```diff
@@ -82,10 +81,6 @@
     "# magic word for producing visualizations in notebook\n",
     "%matplotlib inline\n",
     "\n",
-    "# install a pip package in the current Jupyter kernel\n",
-    "#import sys\n",
-    "#!{sys.executable} -m pip install cvxpy\n",
-    "\n",
     "# import convex solver\n",
     "import cvxpy as cp"
    ]
```
Comment (kshitijc) [inline examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb:86]: I think the previous output object (name: stdout, output_type: stream) containing the text "Requirement already satisfied ... should no longer be needed.

**Review by kshitijc — DISMISSED** [review]

File: examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb (hunk 2 above)
Comment (virajvchaudhari) [reply examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb:72]: Removed, Thanks!

File: examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb [hunk 3]
```diff
@@ -903,10 +884,13 @@
...
  ],
  "metadata": {
+  "interpreter": {
+   "hash": "88fa568089e50b9e3b8fc07b1f035315d2ef539d9a4b33ba4bd65e10c3c73513"
+  },
   "kernelspec": {
    "display_name": "conda_braket",
    "language": "python",
-   "name": "conda_braket"
+   "name": "python3"
```
Comment (kshitijc) [inline examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb:893]: Is this change intentional ?

File: examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb [hunk 4]
```diff
@@ -903,10 +884,13 @@
   }
  ],
  "metadata": {
+  "interpreter": {
+   "hash": "88fa568089e50b9e3b8fc07b1f035315d2ef539d9a4b33ba4bd65e10c3c73513"
+  },
```
Comment (kshitijc) [inline examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb:889]: Is this needed ?

File: examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb [hunk 5]
```diff
@@ -918,9 +902,9 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.10"
+   "version": "3.7.12"
```
Comment (kshitijc) [inline examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb:905]: Should this change ?

Comment (virajvchaudhari) [reply examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb:905]: All the above changes are due to notebook opened locally in IDE. I don't think these should matter much. We have other notebooks with similar changes: https://github.com/aws/amazon-braket-examples/blob/main/examples/braket_features/Verbatim_Compilation.ipynb

**Review by kshitijc — APPROVED** [review]
